ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcService.java

```diff
@@ -82,8 +103,10 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
-  private final Server server;
+  private final Set<Server> servers = new LinkedHashSet<>(3);
```

Comment:
The Server class does not override hashCode and equal so that the behavior may be unexpected.  Let's use a prot-to-Server Map instead?

Reply:
My intention was to rely on reference equality here.  But since items are added to `servers` only if they are different (using the `separateAdminServer` etc. flags), we don't really need a `Set`, so I changed it to a `List`.